lib/everypolitician/popolo/collection.rb

```diff
@@ -18,6 +18,18 @@ def -(other)
         documents.reject { |d| other_ids.include?(d.id) }
       end
 
+      def find_by(attributes = {})
+        where(attributes).first
+      end
+
+      def where(attributes = {})
+        find_all do |object|
+          !attributes.collect do |k,v|
+            object.send(k) == v
+          end.include?(false)
```

Comment:
I had to read this a couple of times to figure out what was going on! Perhaps these three lines would be a bit more readable and concise if it used `all?` instead?

```ruby
attributes.all? { |k, v| object.send(k) == v }
```